File: app/src/main/java/com/example/owner_pc/animechecker/AnimeCheckerApplication.java
package com.example.owner_pc.animechecker;

import android.app.Application;
import android.util.Log;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by owner-PC on 2017/05/15.
 */

public class AnimeCheckerApplication extends Application {
    private Retrofit retrofit;
    private AniListService aniListService;

    @Override
    public void onCreate() {
        super.onCreate();
        // どのActivityからでもAPIを利用できるように、このクラスでAPIを利用する
        setupAPIClient();
    }

    private void setupAPIClient() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor(new HttpLoggingInterceptor.Logger() {
            @Override
            public void log(String message) {
                Log.d("API LOG", message);
            }
        });

        logging.setLevel(HttpLoggingInterceptor.Level.BASIC);

        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(logging).build();

        Gson gson = new GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                .create();

        retrofit = new Retrofit.Builder()
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .baseUrl("https://anilist.co/api/")
                .addConverterFactory(GsonConverterFactory.create(gson))
                .client(client)
                .build();

        aniListService = retrofit.create(AniListService.class);
    }

    public AniListService getAniListService() {
        return aniListService;
    }
}

File: app/src/main/java/com/example/owner_pc/animechecker/view/AnimeCheckerApplication.kt
package com.example.owner_pc.animechecker.view

import android.app.Application
import android.util.Log
import com.example.owner_pc.animechecker.model.AniListService
import com.google.gson.FieldNamingPolicy
import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by owner-PC on 2017/05/15.
 */

class AnimeCheckerApplication : Application() {
    private var retrofit: Retrofit? = null
    var aniListService: AniListService? = null
        private set

    override fun onCreate() {
        super.onCreate()
        // どのActivityからでもAPIを利用できるように、このクラスでAPIを利用する
        setupAPIClient()
    }

    private fun setupAPIClient() {
        val logging = HttpLoggingInterceptor(HttpLoggingInterceptor.Logger { message -> Log.d("API LOG", message) })

        logging.level = HttpLoggingInterceptor.Level.BASIC

        val client = OkHttpClient.Builder().addInterceptor(logging).build()

        val gson = GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                .create()

        retrofit = Retrofit.Builder()
                .baseUrl("https://anilist.co/api/")
                .addConverterFactory(GsonConverterFactory.create(gson))
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(client)
                .build()

        aniListService = retrofit!!.create(AniListService::class.java)
    }
}